tcf_website/views/review.py

```diff
@@ -83,7 +73,32 @@ def new_review(request):
     return render(request, 'reviews/new_review.html')
 
 
+@login_required()
+def check_duplicate(request):
+    """Check for duplicate reviews when a user submits a review
+    based on if it's the same course with the same instructor/semester.
+    Used for an ajax request in new_review.html"""
+
+    form = ReviewForm(request.POST)
+    instance = form.save(commit=False)
+
+    reviews_on_same_class = request.user.review_set.filter(
+        course=instance.course,
+        instructor=instance.instructor,
+        semester=instance.semester
```

Comment:
I still think at the very _least_ we shouldn't allow submitting for the same class/instructor pair on different semesters. I think there's also a compelling case for removing instructors from this filter as well, but that's at least debatable. @Bartleby2718 my interpretation of the example code snippet you posted was just a format to do the filtering instead of taking a stance on this matter; if this wasn't the case, can you clarify?

Reply:
I don't have a strong opinion on this. I was just suggesting what could be potentially more readable code.